deploy/kubernetes/images/prow-canary-sidecar/image.yaml

```diff
@@ -3,7 +3,7 @@ kind: ImageTagTransformer
 metadata:
   name: imagetag-gcepd-driver-prow-head
 imageTag:
-  name: gke.gcr.io/gcp-compute-persistent-disk-csi-driver
+  name: k8s.gcr.io/cloud-provider-gcp/gcp-compute-persistent-disk-csi-driver
```

Comment:
so this is a bug fix. the original transformation from `gke.gcr.io/gcp-compute-persistent-disk-csi-driver` to `gcr.io/k8s-staging-cloud-provider-gcp...` was not working. since prow-gke-release-staging-head was based on [stable-master](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/deploy/kubernetes/images/stable-master/image.yaml#L54) which uses `k8s.gcr.io/cloud-provider-gcp/gcp-compute-persistent-disk-csi-driver`.

Reply:
Correct. Back in January the pd-master-sidecars-latest-k8s-master-on-gce tests were failing because of this so it has been known this needed to be fixed, but I just thought I would do it in this PR.